opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/b3_format.py

```diff
@@ -103,8 +102,9 @@ def extract(
             self._trace_id_regex.fullmatch(trace_id) is None
             or self._span_id_regex.fullmatch(span_id) is None
         ):
-            trace_id = generate_trace_id()
-            span_id = generate_span_id()
+            ids_generator = trace.RandomIdsGenerator()
```

Comment:
Any reason to explicitly instantiate an id generator here instead of using the one that's already configured?

Reply:
Spoke to you offline and you're right that we should be referencing the _global_ `TracerProvider` to find the `IdsGenerator` implementation.